tests/unit/fem/test_linearform_ext.cpp

```diff
@@ -0,0 +1,152 @@
+      REQUIRE(mesh.attributes.Size() == 2);
+      elem_marker.SetSize(2);
+      elem_marker[0] = 0;
+      elem_marker[1] = 1;
+
+      if (problem == DomainLF)
+      {
+         lfi_dev = new DomainLFIntegrator(cst_coeff);
+         lfi_std = new DomainLFIntegrator(cst_coeff);
+      }
+      else if (problem == DomainLFGrad)
+      {
+         lfi_dev = new DomainLFGradIntegrator(dim_cst_coeff);
+         lfi_std = new DomainLFGradIntegrator(dim_cst_coeff);
+      }
+      else if (problem == VectorDomainLF)
+      {
+         lfi_dev = new VectorDomainLFIntegrator(vdim_fct_coeff);
+         lfi_std = new VectorDomainLFIntegrator(vdim_fct_coeff);
+      }
+      else if (problem == VectorDomainLFGrad)
+      {
+         lfi_dev = new VectorDomainLFGradIntegrator(vdim_dim_cst_coeff);
+         lfi_std = new VectorDomainLFGradIntegrator(vdim_dim_cst_coeff);
+      }
+      else { REQUIRE(false); }
+
+      lfi_dev->SetIntRule(gll ? irGLL : ir);
+      lfi_std->SetIntRule(gll ? irGLL : ir);
+
+      lf_dev.AddDomainIntegrator(lfi_dev, elem_marker);
+      lf_std.AddDomainIntegrator(lfi_std, elem_marker);
+   }
+
+   void Run()
+   {
+      const bool scalar = problem == LinearFormExtTest::DomainLF ||
+                          problem == LinearFormExtTest::DomainLFGrad;
+      REQUIRE((!scalar || vdim == 1));
+
+      const bool grad = problem == LinearFormExtTest::DomainLFGrad ||
+                        problem == LinearFormExtTest::VectorDomainLFGrad;
+
+      mfem::out << "[LinearFormExt] " << dim << "D " << mesh_filename
+                << " p=" << p << " q=" << q
+                << (ordering == Ordering::byNODES ? " byNODES " : " byVDIM  ")
+                << vdim << "-"
+                << (scalar ? "Scalar" : "Vector") << (grad ? "Grad" : "Eval")
+                << std::endl;
+
+      const bool use_device = true;
+      lf_dev.Assemble(use_device);
+
+      const bool dont_use_device = false;
+      lf_std.Assemble(dont_use_device);
+
+      REQUIRE(lf_dev*lf_dev == MFEM_Approx(lf_std*lf_std, abs_tol, rel_tol));
+
+      lf_std -= lf_dev;
+      REQUIRE(0.0 == MFEM_Approx(lf_std*lf_std, abs_tol, rel_tol));
+   }
+};
+
+TEST_CASE("Linear Form Extension", "[LinearformExt], [CUDA]")
+{
+   const bool all = launch_all_non_regression_tests;
+
+   const auto mesh =
+      all ? GENERATE("../../data/star.mesh", "../../data/star-q3.mesh",
+                     "../../data/fichera.mesh", "../../data/fichera-q3.mesh") :
+      GENERATE("../../data/star-q3.mesh", "../../data/fichera-q3.mesh");
+   const auto p = all ? GENERATE(1,2,3,4,5,6) : GENERATE(1,3);
+   const auto gll = GENERATE(0,1);
+
+   SECTION("Scalar")
+   {
+      const auto problem = GENERATE(LinearFormExtTest::DomainLF,
+                                    LinearFormExtTest::DomainLFGrad);
+      LinearFormExtTest(mesh, 1, Ordering::byNODES, gll, problem, p).Run();
+   }
+
+   SECTION("Vector")
+   {
+      const auto vdim = all ? GENERATE(1,5,7,24) : GENERATE(1,5);
```

Comment:
The `vdim` here might be overkill. Is there a reason you test such high dimensions?

Reply:
`done`